src/nupic/math/ArrayAlgo.hpp

```diff
@@ -5075,7 +4976,7 @@ namespace nupic {
 
     // See comments in count_gt. We need both conditional compilation and 
     // SSE_LEVEL check.
-#if defined(NTA_OS_LINUX) || defined(NTA_OS_DARWIN)
+#if not defined(NTA_OS_WINDOWS)
```

Comment:
There are a handful of cases where you change from `defined(NTA_OS_LINUX) || defined(NTA_OS_DARWIN)` to `not defined(NTA_OS_WINDOWS)`. I think it is better to condition the code blocks on specific platforms rather than anything-but-windows. That way if there is some other platform we don't accidentally have it use code not intended for it.